### PR TITLE
Minor enhancements to LSEQ implementation and API

### PR DESCRIPTION
**Pull request by bochaco**

### CODE REVIEW

**Comment by bochaco** [comment]
@davidrusu I think it's ready now for reviewing it, they are just minor enhancements and allowing minor customisation to the tree size, plus an `append` function for user's convenience.

File: src/vclock.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ use crate::{Actor, Causal, CmRDT, CvRDT, Dot};
 /// It can tell you if something causally descends something else,
 /// or if different replicas are "concurrent" (were mutated in
 /// isolation, and need to be resolved externally).
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Ord)]
```
Comment (davidrusu) [inline src/vclock.rs:35]: Ah wait I don't think we can do this in general, {A:1} is not comparable to {B:2}. We only get a partial order over VClocks.

File: src/dot.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@ use serde::{Deserialize, Serialize};
 use crate::quickcheck::{Arbitrary, Gen};
 
 /// Dot is a version marker for a single actor
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, Ord)]
```
Comment (davidrusu) [inline src/dot.rs:9]: Why do you want to order Dots? In the general case we don't treat actor ids as comparable so for the case `Dot { actor: "a", counter: 2 }` and `Dot {actor: "b": counter: 2}` we can't order these two dots

**Comment by davidrusu** [comment]
@bochaco w.r.t. the `Ord` impl on the `Dot` and `VClock`, is the goal to have a total ordering over events?
I think we'll need a new structure to encode those semantics. VClocks are used to tell you if there is a conflict between two documents as in this example here:
https://github.com/rust-crdt/rust-crdt/blob/master/examples/vclock.rs#L49

**Comment by bochaco** [comment]
@davidrusu I was also hesitating if those traits would be good to add, the only reason I was adding them just because the way we use instance of this structs require those traits, let me research deeper to fully understand if we can remove such requirements.

**Comment by bochaco** [comment]
I just pushed a commit to remove the Ord derive, it seems we actually don't need that anyways, let me know what you think now.

**Pull request merged**